### PR TITLE
feat: validate new contact method action

### PR DESCRIPTION
**Pull request by getusha**
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/423805
PROPOSAL:

### Tests

1. Go to Settings
2. Go to Profile
3. Go to Contact methods
4. Click “Add new contact method”
5. Enter new email or phone number
6. Verify that you are navigated to magic code verification page
7. Once the magic code is inserted, click Submit
8. Verify that the contact is added and you're navigated to contact list page

- [ ] Verify that no errors appear in the JS console

### Offline tests

1. Go to Settings
2. Go to Profile
3. Go to Contact methods
4. Click “Add new contact method”
5. Verify that the "Add" button is disabled

### QA Steps

**NOTE**: The magic code verification does not work yet so no matter what magic code number you put in, it will succeed 

1. Go to Settings
2. Go to Profile
3. Go to Contact methods
4. Click “Add new contact method”
5. Enter new email or phone number
6. Verify that you are navigated to magic code verification page
7. Once the magic code is inserted, click Submit
8. Verify that the contact is added and you're navigated to contact list page

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/c48868ac-cf1e-495c-bf43-c54919b2c3de

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/f8e9c640-78bc-471d-9b84-7f1ed6c48435

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/e547e9a1-399f-4918-a33a-6984b1070365

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/2a0107d5-0643-4cb8-8334-9691e810cc49

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/c21f9f53-a78b-4821-8bab-53dd4541a9c0

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/c6dc9a03-bda9-4f2d-b929-39d7c683982e

</details>

### CODE REVIEW

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx [hunk 1]
```diff
@@ -47,7 +48,8 @@ function NewContactMethodPage({loginList, route}: NewContactMethodPageProps) {
             const validateIfnumber = LoginUtils.validateNumber(phoneLogin);
             const submitDetail = (validateIfnumber || values.phoneOrEmail).trim().toLowerCase();
 
-            User.addNewContactMethodAndNavigate(submitDetail, route.params?.backTo);
+            User.saveNewContactMethodAndValidateAction(submitDetail);
```
Comment (mountiny) [inline src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx:51]: I think this is clearer as we do not validated it yet 
```suggestion
            User.saveNewContactMethodAndRequestValidationCode(submitDetail);
```

File: src/libs/actions/User.ts [hunk 2]
```diff
@@ -339,12 +414,16 @@ function addNewContactMethodAndNavigate(contactMethod: string, backTo?: string)
...
+            key: ONYXKEYS.ACCOUNT,
+            value: {isLoading: false},
+        },
     ];
 
-    const parameters: AddNewContactMethodParams = {partnerUserID: contactMethod};
+    const parameters: AddNewContactMethodParams = {partnerUserID: contactMethod, validateCode};
 
     API.write(WRITE_COMMANDS.ADD_NEW_CONTACT_METHOD, parameters, {optimisticData, successData, failureData});
-    Navigation.goBack(ROUTES.SETTINGS_CONTACT_METHODS.getRoute(backTo));
```
Comment (mountiny) [inline src/libs/actions/User.ts:347]: Lets rename the method now that we do not navigate

**Comment by hungvu193** [comment]
Thanks for assigning. I'll review this shortly

**Comment by hungvu193** [comment]
Bug 1: User can spam `Didn't receive magic code` button,I think we should also implement countdown timer like login page

https://github.com/user-attachments/assets/0aac1c36-60c9-4076-8f78-dc17f4235cb3

**Comment by getusha** [comment]
>Bug 1: User can spam Didn't receive magic code button,I think we should also implement countdown timer like login page

Same as `ContactMethodDetailsPage` i think we should handle this separately.

**Comment by hungvu193** [comment]
Bug 2: If you added a contact method before but not enter the magic code then exist the page, later when you press `Add new contact method` you will be redirected to `ValidateContactActionPage` page.

https://github.com/user-attachments/assets/2e261d8d-d08d-4b20-bb92-94cb32469047

**Comment by hungvu193** [comment]
Bug 3: A RBR will appear and can not be removed on the primary contact method.

Steps:
1. Click add new contact method.
2. Add any email, press Next
3. Without enter any magic code input, submit the form few times.
4. Notice that a RBR appear right next to the primary contact method.

Screenshot 2024-08-30 at 21 42 00

**Comment by getusha** [comment]
> I've never received any magic code even when I requested/ resent few times. Did you guys receive magic code input after ResendValidateCode was called?

Just did, make sure the email address you are trying to link doesn't have associated accounts.

Screenshot 2024-08-30 at 5 50 22 in the afternoon

**Comment by hungvu193** [comment]
Bug 4: Click on close icon does not clear the error message.

https://github.com/user-attachments/assets/0570f651-ef04-4e8b-9fdd-2c72f999d540

**Comment by hungvu193** [comment]
Bug 5: If you interrupt the api called on `New Contact Method` page, the loading indicator will be there forever, also even if it's loading we can still interact with it.

https://github.com/user-attachments/assets/ad5e8774-a8ee-485d-89b4-f30a999a7cb4

**Comment by hungvu193** [comment]
Oh shoot :/  I got blocked because I requested so many times 😭 . I'll wait for a while then continue with my testing 😈

**Comment by getusha** [comment]
> Oh shoot :/ I got blocked because I requested so many times 😭 . I'll wait for a while then continue with my testing 😈

I use a different email when that happens 😄

**Comment by hungvu193** [comment]
Bug 6: New contact method page shows errors even there's no error.

Steps :
1. Click add new contact method.
2. Add any email, press Next
3. Without entering any magic code, submit the form and request a new code a few times, until you saw the error `Failed to send magic code...`. Click back.
4. Notice that on New Contact Method page, the error shows at the bottom.

https://github.com/user-attachments/assets/3be5a499-7952-4049-af44-0bf0a31da5d6

**Comment by getusha** [comment]
> Bug 2, bug 3, bug 4 (you should be able to see the error on the email form), bug 5

@hungvu193 could you recheck ^ bugs, now? thanks!

**Comment by hungvu193** [comment]
Sure

**Comment by hungvu193** [comment]
I still can reproduce the RBR bug by trying to request magic code few time at `ValidateContactActionPage` then go back to `New Contact method` page then press Next. Repeat few times.

https://github.com/user-attachments/assets/7a0fdb96-4b4d-429e-91e1-3f864ea331f3

File: src/pages/settings/Profile/Contacts/ValidateContactActionPage.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    useEffect(() => {
+        if (!loginData || !!loginData.pendingFields?.addedLogin) {
+            return;
+        }
+
+        // Navigate to methods page on successful magic code verification
+        Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS.route);
+    }, [loginData, loginData?.pendingFields, loginList]);
+
+    useEffect(() => () => User.clearUnvalidatedNewContactMethodAction(), []);
```
Comment (hungvu193) [inline src/pages/settings/Profile/Contacts/ValidateContactActionPage.tsx:35]: Once user existed the page, we should also clear the error (RBR) as well.
Let's also add `User.clearUnvalidatedNewContactMethodAction()` as the return function.

Comment (getusha) [reply src/pages/settings/Profile/Contacts/ValidateContactActionPage.tsx]: Which function are you referring?

**Comment by getusha** [comment]
>I still can reproduce the RBR bug by trying to request magic code few time at ValidateContactActionPage then go back to New Contact method page then press Next. Repeat few times.

@hungvu193 am i doing it right?

https://github.com/user-attachments/assets/c6cc414f-6164-4996-bf49-18ab072c2fda

**Comment by hungvu193** [comment]
Here's the detailed steps for you, don't exist `New Contact Screen`.

https://github.com/user-attachments/assets/3836029a-267b-4f2e-aae9-769ea767d3f3

File: src/pages/settings/Profile/Contacts/ValidateContactActionPage.tsx (hunk 3 above)
Comment (hungvu193) [reply src/pages/settings/Profile/Contacts/ValidateContactActionPage.tsx]: Sorry. I mean the cleanup function of this useEffect.

**Comment by hungvu193** [comment]
It's getting late here, let me know once you complete with the code, I'll take a look when I wake up in the morning 😄

**Comment by hungvu193** [comment]
> Here's the detailed steps for you, don't exist `New Contact Screen`.
> 
>  Screen.Recording.2024-08-31.at.00.11.50.mov

So the problem here is when we called `requestContactMethodValidateCode`, if there's  an error, we add it here:
https://github.com/Expensify/App/blob/117b96112b4b2de8ead192c9c1da619edb9860fa/src/libs/actions/User.ts#L163-L165

Which later makes `UserUtils.getLoginListBrickRoadIndicator(loginList);` return true. Seems we need to add more conditions to fix this case.

**Comment by getusha** [comment]
@hungvu193 Fixed

https://github.com/user-attachments/assets/ed127245-9bc6-4119-9043-fd3e1dd63adf

**Comment by hungvu193** [comment]
@getusha Nice I think it's fixed, how about bug 5?

https://github.com/user-attachments/assets/36bed1c2-41fe-4ead-b79f-8769437ee8da

**Comment by getusha** [comment]
@hungvu193 this is what i see trying to reproduce bug 5

https://github.com/user-attachments/assets/b8ef11e5-3667-44b8-b082-c3401c04c053

**Comment by hungvu193** [comment]
Oh weird, after loging out and login again, I can't reproduce anymore :)

**Comment by hungvu193** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/27622901-9d1d-4d10-a8fd-785a5cb60069


</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by hungvu193** [comment]
On mSafari, after entering the magic code, it still stay on `ValidateContactActionPage`. I tested it with Strict mode enabled.

https://github.com/user-attachments/assets/ad3db492-2333-4223-b6c5-ff5d516a9a07

**Comment by getusha** [comment]
>On mSafari, after entering the magic code, it still stay on ValidateContactActionPage. I tested it with Strict mode enabled.

@hungvu193 could you logout & sign in again then retest?

**Comment by hungvu193** [comment]
> > On mSafari, after entering the magic code, it still stay on ValidateContactActionPage. I tested it with Strict mode enabled.
> 
> @hungvu193 could you logout & sign in again then retest?

I actually logged out before testing, but let me try again.

**Comment by hungvu193** [comment]
Weird, I still can reproduce. 

https://github.com/user-attachments/assets/9d7d887c-95a7-4468-97ed-e56c390e2cfe

**Review by hungvu193 — APPROVED** [review]
All yours @mountiny

**Review by mountiny — APPROVED** [review]
@getusha Thank you!

I think we will need to add some clean ups but for the sake of the urgency I think this is good to go and I would like to merge it before the deploy so we have it in staging tomorrow to unblock the backend changes

@getusha Could you look into the fixing the navigation when the user removes the optimistic login? The page goes to not found page because the contact method is in the URL. I think we need to redirect to the contact method list page

https://github.com/user-attachments/assets/6f6e1f28-d892-4ec2-acae-f248eb9728e2

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx [hunk 4]
```diff
@@ -38,19 +39,29 @@ function NewContactMethodPage({loginList, route}: NewContactMethodPageProps) {
...
+        User.saveNewContactMethodAndRequestValidationCode(submitDetail);
+    }, []);
+
+    useEffect(() => {
+        if (!pendingContactAction?.validateCodeSent) {
+            return;
+        }
+
+        Navigation.navigate(ROUTES.SETINGS_CONTACT_METHOD_VALIDATE_ACTION);
+    }, [pendingContactAction]);
```
[the remark this answers is not in the document]
Comment (getusha) [reply src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx:62]: @parasharrajat We started requiring verification before adding a contact method.  the new flow is `temporarily save new contact > verify action > add the new contact method.`

`saveNewContactMethodAndRequestValidationCode` just saves the contact method temporarily and requests a magic code, so this logic just navigates you`ValidateContactActionPage` **after** the magic code is sent

https://github.com/Expensify/App/blob/e1d726e7f5bc0b3e1b5c24ad088411528f2d2a57/src/libs/actions/User.ts#L342

**Comment by getusha** [comment]
@parasharrajat there is no design doc but you can check this [thread](https://expensify.slack.com/archives/C02NK2DQWUX/p1724862676925179) and this [issue](https://github.com/Expensify/App/issues/48541) for more detail

**Comment by mountiny** [comment]
No design doc for this, its an urgent change we require now to validate the primary login before taking certain actions, little more details in here https://github.com/Expensify/App/issues/48541
